server/src/main/java/keywhiz/service/resources/automation/v2/LogArgumentsImpl.java

```diff
@@ -0,0 +1,23 @@
+package keywhiz.service.resources.automation.v2;
+
+import com.google.common.collect.Streams;
+import java.lang.reflect.Parameter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LogArgumentsImpl implements MethodInterceptor {
+  private static final Logger logger = LoggerFactory.getLogger(LogArguments.class);
+  public Object invoke(MethodInvocation invocation) throws Throwable {
+    Parameter[] parameters = invocation.getMethod().getParameters();
+    Object[] arguments = invocation.getArguments();
+    Stream<Map.Entry<Parameter, Object>> parameterArgumentPairs = Streams.zip(Arrays.stream(parameters), Arrays.stream(arguments), Map::entry);
+    logger.info(String.format("Function Name: %s, Function Arguments: %s",invocation.getMethod().getName(),
+        Arrays.deepToString(parameterArgumentPairs.toArray())));
```

Comment:
Can potentially group together L16-L18 and L20

Reply:
If I'm reading this correctly, the map operation on line 16 will produce either the parameter name (type String) or the parameter object itself (type Parameter), and the toString() on Parameter includes the modifiers, type, name, and some other info according to the docs:

https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Parameter.html#toString--

If so, I think that's going to really blow up the log statement in the case where we don't have parameters available (which shouldn't ever really be the case given we control the compiler settings as well, but...).

I was thinking more of an iterative approach, like:

```
for (int i = 0; i < copyOfArguments.length; i++) {
  if (parameters[i].isNamePresent()) {
    copyOfArguments[i] = Map.entry(parameters[i].getName(), copyOfArguments[i]);
  }
}
```

I know we discussed trying to grab argument names from the various annotations on the parameters as a possible backup, but if you were planning on adding that in a future revision I think we could skip that for now since this approach is pretty awesome by itself. Fancy stuff!